### ISSUE TITLE
In case of failure, logs should be saved as artefact

### ISSUE THREAD

**Issue by Tchoupinax** [iggy-rs/setup-iggy#15]
Logs are written inside `local_data/logs` by default.
Also, it can be overridden with `IGGY_SYSTEM_PATH`.

**Comment by hubcio** [comment]
Make it an option, like we discussed:
- `always` upload logs as an artifact (compressed ofc)
- `never` upload logs from iggy-server to github, regardless of action status
- `on-failure` - upload logs when action fails for any reason (this action OR whole flow?)

Feel free to modify those or add any other options.

**Comment by Tchoupinax** [comment]
Hey!
https://github.com/iggy-rs/setup-iggy/pull/23 does the job however conditional behavior is not yet done.

- It would be hard (or impossible) to detect iggy errors in this Github Actions because the server is started in background and then we continue to the next step. If the server fails, action is not aware because process is detached. The only seen consequence is that test will fail later.
- `success()` and `failure()` native function works well. I tested to include them conditionally with input but I did not find the correct way. 

Regarding these two points, I decided to upload logs every time. It's not a big deal and an acceptable trade-off. We will see later with usage where we want to go!
